bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt

```diff
@@ -0,0 +1,112 @@
+package social.bigbone.api.entity
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import social.bigbone.DateTimeSerializer
+import social.bigbone.PrecisionDateTime
+import social.bigbone.api.entity.Role.Permission
+
+/**
+ * Represents a custom user role that grants permissions.
+ *
+ * @see <a href="https://docs.joinmastodon.org/entities/Role/">Mastodon API Role</a>
+ */
+@Serializable
+data class Role(
+    /**
+     * The ID of the Role in the database.
+     */
+    @SerialName("id")
+    val id: Int,
+
+    /**
+     * The name of the role.
+     */
+    @SerialName("name")
+    val name: String,
+
+    /**
+     * The hex code assigned to this role.
+     * If no hex code is assigned, the string will be empty.
+     */
+    @SerialName("color")
+    val color: String,
+
+    /**
+     * A bitmask that represents the sum of all permissions granted to the role.
+     * Possible values can be found in [Permission].
+     *
+     * For convenience, use [getParsedPermissions] to directly get a list of [Permission]s seen in this field.
+     */
+    @SerialName("permissions")
+    val rawPermissions: Int,
+
+    /**
+     * Whether the role is publicly visible as a badge on user profiles.
+     */
+    @SerialName("highlighted")
+    val highlighted: Boolean,
+
+    /**
+     * When the role was first assigned.
+     */
+    @SerialName("created_at")
+    @Serializable(with = DateTimeSerializer::class)
+    val createdAt: PrecisionDateTime = PrecisionDateTime.InvalidPrecisionDateTime.Unavailable,
+
+    /**
+     * When the role was last updated.
+     */
+    @SerialName("updated_at")
+    @Serializable(with = DateTimeSerializer::class)
+    val updatedAt: PrecisionDateTime = PrecisionDateTime.InvalidPrecisionDateTime.Unavailable
+) {
+
+    /**
+     * Returns whether this role grants the [Permission] specified in [permission].
+     * The Mastodon API returns permissions as a bitmask in [rawPermissions].
+     */
+    @Suppress("DataClassContainsFunctions")
```

Comment:
I’ve added this suppression as I don’t agree with detekt’s defaults here.
I want these essentially conversion methods to be limited to this class’ instances only so I think the current way is the best we could do.